### DISCUSSION TITLE
How do you keep the version number in sync between tauri's config file and Cargo.toml?

### DISCUSSION THREAD

**Discussion by beeb** [tauri-apps/tauri#6347] [Q&A] [answered]
At first I thought I would have to manage 3 versions (the root `package.json`, the `tauri.conf.json` and the `Cargo.toml`), but I noticed that the root `package.json` doesn't require a version number, so I removed it.

However, cargo requires a version to be defined in its manifest. How do you keep that version number in sync with Tauri's config file? Does it need to be in sync? Can you remove the one in Tauri's config file and would it default to the one in the `Cargo.toml` file?

**Comment by beeb** [comment, marked as answer]
I should have tried for myself, it seems indeed that, if no value is provided for the package name and version in `tauri.conf.json`, it will default to the values for the crate defined in `Cargo.toml`. So `Cargo.toml` will be my single source of truth. Sweet!

**Comment by soulflyman** [comment]
I came here by a web search and leave this information because it could help someone else.
The [documentation ](https://v2.tauri.app/reference/config/#version) reads:
```
App version. It is a semver version number or a path to a package.json file containing the version field.

If removed the version number from Cargo.toml is used. It’s recommended to manage the app versioning in the Tauri config.
```

What I have not found out is if the version information in the Cargo.toml is of any relevance. Is it a drop? Is it overwritten with the value from tauri.conf? Are both values written to the binary somewhere if mismatched?

**Reply by FabianLars** [reply]
> Is it overwritten with the value from tauri.conf?

Yes. It is only used when you do not set the version in tauri.conf. The only place where the Cargo.toml version will show up is in the cargo build logs.
